lib/rubocop/cop/lint/or_assignment_to_constant.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks unintended or-assignment to constant.
+      # Usually, you don't need to consider whether A is already assigned elsewhere.
+      #
+      # @example EnforcedStyle: bar (default)
+      #
+      #   # bad
+      #   CONST ||= 1
+      #
+      #   # good
+      #   CONST = 1
+      #
+      class OrAssignmentToConstant < Base
+        extend AutoCorrector
+
+        MSG = 'Probably unintended or-assignment. Use plain assignment.'
```

Comment:
```suggestion
        MSG = 'Avoid use or-assignment with constants.'
```

Might be useful to include the correction in the message as well (<code>Use &#x60;...&#x60; instead.</code>)

Reply:
I applied the commit-suggestion.